index.html

```diff
@@ -147,12 +156,31 @@ <h3><dfn>CropTarget</dfn> Definition</h3>
           purpose is to be handed to {{BrowserCaptureMediaStreamTrack/cropTo}} as input.
         </p>
         <pre class="idl">
-          [Exposed = Window]
+          [Exposed = Window, Serializable]
           interface CropTarget {
             // Intentionally empty; just an opaque identifier.
           };
         </pre>
         <dl data-link-for="CropTarget" data-dfn-for="CropTarget"></dl>
+
+        <p id="serializable-croptarget">
+          {{CropTarget}} objects are [= serializable objects =] [[!HTML]]. Their [= serialization
+          steps =], given <var>value</var> and <var>serialized</var>, are:
```

Comment:
"The introduction of these steps should omit mention of the forStorage argument if it is not relevant to the algorithm."

Reply:
What is the use case for forStorage being set to true?

Reply:
Do you want this to be able to end up in IDB/pushState/some other things or not, essentially.

Reply:
I want it to be possible for this to be stringified and sent over a REST call, as I know some Web applications that can cooperate either in a single-tab or multi-tab setting, and they use REST (and REST-like) for as much as they can, since that's already set up and works for them. I think it's a valid use-case and there's no reason to forbid it.

Reply:
You mean through Fetch? Serializable doesn't help with that.

Reply:
If Serializable forStorage=true does not allow conversion to a string, could you please advise what does?

Reply:
If you want to convert a platform object to a string, you need to define a stringifier for it. That seems like a topic for a new issue though as #21 doesn't cover it.

Reply:
* I've added a slot for [[Data]].
* Indeed, stringification is a separate issue. If you happen to know the mechanism, I'd be happy to hear it. At any rate, we can do so as a follow-up.

Reply:
> > Looking at the new PR, I think it is missing the forStorage handling, something like:
> > 
> > * Add forStorage as parameter to the serialisation steps algorithm.
> > * As first step of the algorithm, throw an exception if forStorage is set to true.

I agree with @youennf here.

> On the PR itself, I have quoted this part:
> 
> > The introduction of these steps should omit mention of the forStorage argument if it is not relevant to the algorithm.
> 
> It's taken from [here](https://whatpr.org/html/6523/structured-data.html#serializable-objects). I read it as "ignore this if it's not relevant."

I think you're misinterpreting it. It says _"The introduction of these steps should omit mention of the forStorage argument if it is not relevant to the algorithm."_ which seems to say to not list the argument in the argument list unless it is referenced in the algorithm. Since the this algorithm would throw on it, that is satisfied.

Cc-ing @annevk as the authority on what the intent as far as default behavior here is, if any.

> Seems irrelevant enough to me. Or why do you think it's important to specify throwing?

We should narrow the scope of functionality to what's necessary for the use cases we're solving. If we don't need to implement storage, we shouldn't. The iterative process for adding functionality, is we hear use cases that demand it, and then discuss adding it.

It's easy to add more functionality later. What's hard is to remove it.